homeassistant/components/google_assistant_sdk/helpers.py

```diff
@@ -56,22 +59,47 @@ class CommandResponse:
     text: str
 
 
-async def async_send_text_commands(
-    hass: HomeAssistant, commands: list[str], media_players: list[str] | None = None
-) -> list[CommandResponse]:
-    """Send text commands to Google Assistant Service."""
-    # There can only be 1 entry (config_flow has single_instance_allowed)
-    entry: ConfigEntry = hass.config_entries.async_entries(DOMAIN)[0]
-
+async def async_create_credentials(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> Credentials:
+    """Create credentials to pass to TextAssistant."""
+    # Credentials already exist in memory, return that.
+    if DATA_CREDENTIALS in hass.data[DOMAIN][entry.entry_id]:
+        return hass.data[DOMAIN][entry.entry_id][DATA_CREDENTIALS]
+
+    # Check if there is a json file created with google-oauthlib-tool with application type of Desktop app.
+    # This is needed for personal results to work.
+    credentials_json_filename = hass.config.path(
+        "google_assistant_sdk_credentials.json"
+    )
+    if os.path.isfile(credentials_json_filename):
+        with open(credentials_json_filename, encoding="utf-8") as credentials_json_file:
+            credentials = Credentials(token=None, **json.load(credentials_json_file))
```

Comment:
Why is a token file needed? The Google Calendar integration has a flow for TV and limited input credentials type. Isn't that an option here?

Reply:
Unfortunately not. That's the only way which requires quite some steps as outlined in the documentation PR.

Reply:
Did you test all the different credential types?